crates/subspace-core-primitives/src/pieces.rs

```diff
@@ -650,6 +650,15 @@ impl Record {
         // SAFETY: `Record` is `#[repr(transparent)]` and guaranteed to have the same memory layout
         unsafe { mem::transmute(value) }
     }
+
+    /// Convert from a record to its raw bytes. Used for object reconstruction.
```

Comment:
I'm okay merging this, but I'm not sure whether such an API here is appropriate because it is not valid for 100% of pieces.

Only source pieces have zero bytes, parity pieces have non-zero bytes most of the time and such a method makes no sense in those cases. Right now this method is only used in tests and for actual data retrieval this will not help much either since there are many more things to consider there, so it is not like you can call this method and ignore the rest of challenges with data layout.

All of it together makes this method of not much help in practice.